client/layout/guided-tours/config-elements.js

```diff
@@ -0,0 +1,158 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import { selectStep } from 'state/ui/guided-tours/selectors';
+
+const bindTourName = ( nextFn, tourName ) => stepName =>
+	nextFn( { tour: tourName, stepName } );
+
+export const Tour = ( { name, context, children, state, next, quit } ) => {
+	console.log( 'tour state ', state );
+	console.log( 'tour props ', next, quit );
+	const nextStep = selectStep( state, children );
+	if ( ! nextStep || ! context( state ) ) {
+		return null;
+	}
+
+	return React.cloneElement( nextStep, {
+		state,
+		quit,
+		next: bindTourName( next, name )
+	} );
+};
+
+Tour.propTypes = {
+	name: PropTypes.string.isRequired
+};
+
+export class Step extends Component {
+	constructor( props ) {
+		super( props );
+		this.next = this.next.bind( this );
+		this.quit = this.quit.bind( this );
+	}
+
+	componentWillMount() {
+		console.log( 'state', this.props.state );
+		this.skipIfInvalidContext( this.props );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		this.skipIfInvalidContext( nextProps );
+	}
+
+	skipIfInvalidContext( props ) {
+		const { context, state } = props;
+		if ( context && ! context( state ) ) {
+			this.next( props );
+		}
+	}
+
+	next( props ) {
+		const { next, nextStep } = props;
+		next( nextStep );
+	}
+
+	quit( props ) {
+		const { quit } = props;
+		quit( /** finished **/ );
+	}
+
+	render() {
+		console.log( this.props );
+		const { context, children, state } = this.props;
+		if ( context && ! context( state ) ) {
+			return null;
+		}
+		return (
+			<div className="guided-tours__step">
+				{ React.cloneElement( children, {
+					next: this.next.bind( this, this.props ),
+					quit: this.quit.bind( this, this.props ),
```

Comment:
We already bind `next` and `quit` to the instance, so… idea: make those methods default to using `this.props`, but have them take an optional `props` argument.